pkg/bpf/map_linux.go

```diff
@@ -117,6 +118,9 @@ type Map struct {
 	// DumpParser is a function for parsing keys and values from BPF maps
 	DumpParser DumpParser
 
+	// withValueCache is true when map cache has been enabled
+	withValueCache bool
```

Comment:
Is this different than `m.cache != nil`? Tracking an extra variable seems like provides an opportunity for the bool vs. the pointer to get out of sync.

If there is an actual difference, that would be useful context to put in the comment above this field.

Reply:
Added comment on `m.cache` member that hopefully clarifies.